sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java

```diff
@@ -570,6 +570,7 @@ private SqlTestFramework(Builder builder)
         // test pulls in a module, then pull in that module, even though we are
         // not the Druid node to which the module is scoped.
         .ignoreLoadScopes()
+        .addModule(binder -> binder.bind(Closer.class).toInstance(resourceCloser))
```

Comment:
What's depending on a `Closer` to be injected?  There should likely be some other way of achieving this.

Reply:
Looks like it's the AvaticaServer.  That should have its lifecycle managed more closely/explicitly.  Just relying on a closer like this generates a pattern in code that leads to resource leaks and poor lifecycle management (I realize that it's probably safe in this one test, but the pattern itself gets seen and then repeated).

You should either mark it as `@LifecycleManaged`, if the test is getting a Lifecycle object, calling `.start()` and `.stop()` on it, then this will end up registering the object with that `Lifecycle` just naturally.  This is the best pattern.

If the test isn't using the `Lifecycle` object, then it should be explicitly getting the `AvaticaServer` instance out of the Injector and closing it explicitly in the `close` method.

Reply:
we've talked about this offline - to fix this by properly closing the object where it was created (this is the current version).
the `LifecycleManaged` is not yet used around here - but it would be much better to rely on that; I'll probably do that in a later changeset - if the need to cleanup configurations will be necessary.